samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c

```diff
@@ -627,6 +651,163 @@ int dect_phy_mac_client_associate(struct dect_phy_mac_nbr_info_list_item *target
+		association_data->target_nbr = target_nbr;
+	}
+	err = dect_phy_mac_client_associate_msg_send(target_nbr, params);
+	if (err) {
+		desh_error("(%s): dect_phy_mac_client_associate_msg_send failed: %d",
+			(__func__), err);
+		return err;
+	}
+	association_data->state = DECT_PHY_MAC_CLIENT_ASSOCIATION_STATE_WAITING_ASSOCIATION_RESP;
+	association_data->target_long_rd_id = params->target_long_rd_id;
+	association_data->target_nbr = target_nbr;
+
+	k_work_init_delayable(
+		&association_data->association_resp_wait_work,
+		dect_phy_mac_client_associate_resp_wait_worker);
+
+	/* Start worker for waiting association response */
+	k_work_schedule_for_queue(&dect_phy_ctrl_work_q,
+				  &association_data->association_resp_wait_work,
+				  K_SECONDS(DECT_PHY_MAC_CLIENT_ASSOCIATION_RESP_WAIT_TIME_SEC));
+	return 0;
+}
+
+void dect_phy_mac_client_nbr_scan_completed_cb(
+	struct dect_phy_mac_nbr_bg_scan_op_completed_info *info)
+{
+	struct dect_phy_mac_client_association_data *association_data =
+		dect_phy_mac_client_association_data_get(info->target_long_rd_id);
+
+	if (!association_data) {
+		desh_warn("(%s): No association data found for target long rd id %u",
+			(__func__), info->target_long_rd_id);
+		return;
+	}
+	if (info->cause == DECT_PHY_MAC_NBR_BG_SCAN_OP_COMPLETED_CAUSE_USER_INITIATED) {
+		desh_print("(%s): Background scan completed for target long rd id %u. "
+			   "Cause: user initiated",
+			(__func__), info->target_long_rd_id);
+	} else {
+		__ASSERT_NO_MSG(info->cause ==
+			DECT_PHY_MAC_NBR_BG_SCAN_OP_COMPLETED_CAUSE_UNREACHABLE);
+		desh_warn("(%s): Background scan completed for target long rd id %u. "
+			   "Cause: unreachable for a %d msecs.",
+			(__func__), info->target_long_rd_id,
+			DECT_PHY_MAC_NBR_BG_SCAN_MAX_UNREACHABLE_TIME_MS);
+	}
+	association_data->bg_scan_on_going = false;
+}
+
+void dect_phy_mac_client_associate_resp_handle(
+	dect_phy_mac_common_header_t *common_header,
+	dect_phy_mac_association_resp_t *association_resp)
+{
+	struct dect_phy_mac_client_association_data *association_data =
+		dect_phy_mac_client_association_data_get(common_header->transmitter_id);
+
+	if (!association_data) {
+		desh_warn("(%s): No association data found for transmitter id %u",
+			(__func__), common_header->transmitter_id);
+		return;
+	}
+	k_work_cancel_delayable(&association_data->association_resp_wait_work);
+
+	association_data->state =  DECT_PHY_MAC_CLIENT_ASSOCIATION_STATE_ASSOCIATED;
+	desh_print("(%s): associated with device %d - starting background scan",
+		(__func__), common_header->transmitter_id);
+
+	struct dect_phy_mac_nbr_bg_scan_params bg_scan_params;
+
+	bg_scan_params.cb_op_completed = dect_phy_mac_client_nbr_scan_completed_cb;
+	bg_scan_params.target_nbr = association_data->target_nbr;
+	bg_scan_params.target_long_rd_id = association_data->target_long_rd_id;
+	bg_scan_params.phy_op_handle = association_data->bg_scan_phy_handle;
+
+	if (association_data->bg_scan_on_going) {
```

Comment:
A stupid question, why is background scan first stopped and then started immediately?

Reply:
Yes